awssql/host_info_util/host_info_util.go

```diff
@@ -32,9 +37,56 @@ func AreHostListsEqual(s1 []*HostInfo, s2 []*HostInfo) bool {
 
 func GetWriter(hosts []*HostInfo) *HostInfo {
 	for _, host := range hosts {
-		if host.Role == WRITER {
+		if host != nil && host.Role == WRITER {
 			return host
 		}
 	}
 	return nil
 }
+
+func GetReaders(hosts []*HostInfo) []*HostInfo {
+	readerHosts := make([]*HostInfo, 0, len(hosts))
+	for _, host := range hosts {
+		if host != nil && host.Role == READER {
+			readerHosts = append(readerHosts, host)
+		}
+	}
+	slices.SortFunc(readerHosts, func(i, j *HostInfo) int {
+		if i.Host < j.Host {
+			return -1
+		} else if i.Host > j.Host {
+			return 1
+		} else {
+			return 0
+		}
```

Comment:
Is this in lexographical order?  If so, can we just do strings.Compare(i.host, j.host)?